lib/matplotlib/backend_bases.py

```diff
@@ -1951,18 +1951,32 @@ def get_supported_filetypes_grouped(cls):
             groupings[name].sort()
         return groupings
 
-    def _get_output_canvas(self, fmt):
+    def _get_output_canvas(self, backend, fmt):
         """
-        Return a canvas suitable for saving figures to a specified file format.
+        Set the canvas in preparation for saving the figure.
 
-        If necessary, this function will switch to a registered backend that
-        supports the format.
-        """
-        # Return the current canvas if it supports the requested format.
-        if hasattr(self, 'print_{}'.format(fmt)):
+        Parameters
+        ----------
+        backend : str or None
+            If not None, switch the figure canvas to the ``FigureCanvas`` class
+            of the given backend.
+        fmt : str
+            If *backend* is None, then determine a suitable canvas class for
+            saving to format *fmt* -- either the current canvas class, if it
+            supports *fmt*, or whatever `get_registered_canvas_class` returns;
+            switch the figure canvas to that canvas class.
+        """
+        if backend is not None:
+            # Return a specific canvas class, if requested.
+            canvas_class = (
```

Comment:
Is it worth checking that this canvas class can actually produce the format requested?

Reply:
I actually think we should error out if someone explicitly requires a backend that is incompatible with the required format.  I would say erroring out with an AttributeError is fine but I can wrap that in something with a better error message if you prefer.

Reply:
I think raising a descriptive error message from here (rather than deeper into `savefig` is the best option.

Reply:
sure, done